source/parser.hera

```diff
@@ -6599,6 +6604,14 @@ TypeAliasDeclaration
       ts: true,
     }
 
+  InsertType IdentifierName:id TypeParameters? __ TypeAssignment ( ( _? Type ) / ( __ Type ) ) ->
```

Comment:
I think we'd ideally check for a nested tyoe here (with PushIndent Type? PopIndent, if there isn't a rule already). This is necessary to set indentation correctly. I think other assignments have a similar rule?

Reply:
The don't currently but they should.

Reply:
Oh, actually they do from `ExtendedExpression` -> ... -> `NestedNonAssignmentExtendedExpression`

Reply:
I think for this we could use `MaybeIndentedType` instead of `( ( _? Type ) / ( __ Type ) )`.

Reply:
I copied `( ( _? Type ) / ( __ Type ) )` from the other formation of `TypeAliasDeclaration`. It's definitely not what I would have come up with on my own.